### PR TITLE
Update eslint to latest version and fix/ignore errors

### PR DESCRIPTION
**Pull request by jglamine**
I've updated eslint to the latest version and updated our config file to match the latest version.

I either fixed or ignored the new errors (there were about 35).

### CODE REVIEW

File: packages/lesswrong/lib/collections/posts/views.js [hunk 1]
```diff
@@ -419,7 +419,7 @@ Posts.addView("afRecentDiscussionThreadsList", terms => {
 
 Posts.addView("legacyPostUrl", function (terms) {
   return {
-    selector: {"legacyData.url": {$regex: "\/lw\/"+terms.legacyUrlId+"\/.*"}},
+    selector: {"legacyData.url": {$regex: "/lw/"+terms.legacyUrlId+"/.*"}},
```
Comment (Discordius) [inline packages/lesswrong/lib/collections/posts/views.js]: Wait, did you test this? This seems exactly like the kind of thing that could break.

**Comment by Discordius** [comment]
The rest seems great. Pleasantly surprised about how little we need to do to move to the new lifecycle methods.

File: packages/lesswrong/lib/collections/posts/views.js (hunk 1 above)
Comment (jglamine) [reply packages/lesswrong/lib/collections/posts/views.js]: I didn't test it on the site, but testing in a JavaScript console:
```
> const terms = { legacyUrlId: "testId" };
  "\/lw\/"+terms.legacyUrlId+"\/.*" === "/lw/"+terms.legacyUrlId+"/.*";
< true
```
The two strings are the same - the backslash escape characters are not included in the actual string that is constructed.

**Comment by jglamine** [comment]
Well, we haven't exactly moved to the new lifecycle methods - just switched to the ones called `UNSAFE`. The "correct" way would be to refactor our code to not use those methods, but that seems outside of the scope of this issues.

**Pull request merged**